src/Tools/ExternalAccess/AspNetCore/Internal/EmbeddedLanguages/AspNetCoreRouteEmbeddedLanguageClassifier.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.EmbeddedLanguages;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.Internal.EmbeddedLanguages
+{
+    [ExportEmbeddedLanguageClassifierInternal(
+        nameof(AspNetCoreRouteEmbeddedLanguageClassifier), LanguageNames.CSharp, supportsUnannotatedAPIs: false, "Route"), Shared]
+    internal class AspNetCoreRouteEmbeddedLanguageClassifier : IEmbeddedLanguageClassifier
```

Comment:
@JamesNK  this is the type we export on your behalf that hooks you into roslyn.  this allows us to change that general shape without breaking you.  This type then imports the IAspNetCoreRouteEmbeddedLanguageClassifier value which you export to defer to you to actually do the work.

if we ever need to change the full end to end communication api, this allows us to stage the individual parts.